From: Daniel van Vugt <daniel.van.vugt@canonical.com>
Date: Tue, 29 Mar 2022 18:08:27 +0800
Subject: clutter/pick-stack: Avoid uninitialized
 cairo_region_subtract_rectangle

This fixes instances of:
```
*** BUG ***
In pixman_region32_init_rect: Invalid rectangle passed
Set a breakpoint on '_pixman_log_error' to debug
```
seen when navigating the overview and launching apps.

Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/2349>

Origin: https://gitlab.gnome.org/GNOME/mutter/-/commit/b2805dd5
---
 clutter/clutter/clutter-pick-stack.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/clutter/clutter/clutter-pick-stack.c b/clutter/clutter/clutter-pick-stack.c
index db5c8d3..e11d248 100644
--- a/clutter/clutter/clutter-pick-stack.c
+++ b/clutter/clutter/clutter-pick-stack.c
@@ -491,7 +491,9 @@ calculate_clear_area (ClutterPickStack  *pick_stack,
 	   rec->base.rect.y1 == rec->base.rect.y2))
         continue;
 
-      clutter_actor_get_paint_box (rec->actor, &paint_box);
+      if (!clutter_actor_get_paint_box (rec->actor, &paint_box))
+        continue;
+
       cairo_region_subtract_rectangle (area,
                                        &(cairo_rectangle_int_t) {
                                          .x = paint_box.x1,
