Index: mutter/src/backends/x11/meta-backend-x11.c
===================================================================
--- mutter.orig/src/backends/x11/meta-backend-x11.c
+++ mutter/src/backends/x11/meta-backend-x11.c
@@ -373,7 +373,6 @@ handle_host_xevent (MetaBackend *backend
         {
           switch (xkb_ev->any.xkb_type)
             {
-            case XkbNewKeyboardNotify:
             case XkbMapNotify:
               keymap_changed (backend);
               break;
